internal/pkg/instrumentation/bpf/net/http/client/probe.go

```diff
@@ -195,21 +196,33 @@ func convertEvent(e *event) []*probe.SpanEvent {
 	}
 
 	proto := unix.ByteSliceToString(e.Proto[:])
+	scheme := ""
 	if proto != "" {
 		parts := strings.Split(proto, "/")
 		if len(parts) == 2 {
+			if parts[0] != "HTTP" {
+				attrs = append(attrs, semconv.NetworkProtocolName(parts[0]))
+			}
+			scheme = strings.ToLower(parts[0])
 			attrs = append(attrs, semconv.NetworkProtocolVersion(parts[1]))
 		}
 	}
 
-	return []*probe.SpanEvent{
-		{
-			SpanName:          method,
-			StartTime:         int64(e.StartTime),
-			EndTime:           int64(e.EndTime),
-			SpanContext:       &sc,
-			Attributes:        attrs,
-			ParentSpanContext: pscPtr,
-		},
+	fullURL := fmt.Sprintf("%s://%s%s", scheme, serverAddr.Value.AsString(), path)
```

Comment:
@RonFed can we get a pointer to the URL struct from the request? https://pkg.go.dev/net/http#Request store that in the Event, and just call `String()` on that?

Reply:
Since that pointer will be to some memory in the instrumented process, we won't be able to read from it here. I think we should add the relevant fields of the URL struct to `offsetgen` and read them similarly to the other fields.

Reply:
Makes sense, they are all pretty straightforward fields except for `Userinfo` which is also a pointer and we'll have to rely on the unexported fields from that struct